packages/wonder-blocks-modal/src/components/modal-panel.tsx

```diff
@@ -105,11 +101,9 @@ export default function ModalPanel({
     const {theme} = useScopedTheme(ModalDialogThemeContext);
     const styles = useStyles(themedStylesFn, theme);
 
-    const renderMainContent = (): React.ReactNode => {
-        const mainContent = ModalContent.isClassOf(content) ? (
-            (content as React.ReactElement<
-                React.ComponentProps<typeof ModalContent>
-            >)
+    const renderMainContent = React.useCallback((): React.ReactNode => {
```

Comment:
**Praise**: I like the use of `useCallback()` to make this more performant.

**No-op question**: Should we be putting a `useCallback()` around all sub-render functions? Is there a preference to break them out into separate components, or does it just depend on context?